files/check_zfs

```diff
@@ -0,0 +1,140 @@
+                        help="Name of the ZFS pool to check. If omitted, the first pool in `zpool list` is checked.",
+                        type=str)
+
+    args = vars(parser.parse_args())
+    # Check if thresholds are in range
+    for arg_key in ["capacity_warn", "capacity_crit", "fragmentation_warn", "fragmentation_crit"]:
+        if args[arg_key] < 0 or args[arg_key] > 100:
+            raiseState('UNKNOWN', ret)
+            ret['output'] = "Value {} for key {} is not in range [0, 100].".format(args[arg_key], arg_key)
+            return
+
+    # List pools
+    zpoolListCommand = [ZPOOL_COMMAND, 'list']
+    if args['pool'] is not None:
+        zpoolListCommand.append(args['pool'])
+    childProcess = subprocess.Popen(zpoolListCommand, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+    zpoolListOutput = childProcess.communicate()[0]
+
+    if childProcess.returncode is not 0:
+        raiseState('UNKNOWN', ret)
+        ret['output'] = '`{}` failed with return code {}.'.format(' '.join(zpoolListCommand), childProcess.returncode)
+        ret['details'] = zpoolListOutput
+        return
+
+    zpoolLines = zpoolListOutput.splitlines()
+    zpoolHeaders = zpoolLines[0].split()
+    zpoolValues = zpoolLines[1].split()
+
+    zpoolState = {}
+    for index, key in enumerate(zpoolHeaders):
+        zpoolState[key.decode('UTF-8')] = zpoolValues[index].decode('UTF-8')
+
+    capPercent = int(zpoolState['CAP'].replace("%", ""))
+    fragPercent = int(zpoolState['FRAG'].replace("%", ""))
+
+    # Calculate performance data
+    ret['perfdata'] += 'capacity={}%;{};{},0,100 '.format(capPercent, args['capacity_warn'],
+                                                          args['capacity_crit'])
+    ret['perfdata'] += 'fragmentation={}%;{};{},0,100 '.format(fragPercent, args['fragmentation_warn'],
+                                                               args['fragmentation_crit'])
+    allocGB = convertToGB(zpoolState['ALLOC'])
+    sizeGB = convertToGB(zpoolState['SIZE'])
+    capWarnGB = sizeGB * args['capacity_warn'] / 100
+    capCritGB = sizeGB * args['capacity_crit'] / 100
+    ret['perfdata'] += 'allocation={}GB;{};{};0;{} '.format(allocGB, capWarnGB, capCritGB, sizeGB)
+
+    ret['perfdata'] += 'deduplication={}'.format(zpoolState['DEDUP'])
+
+    # Check state
+    if zpoolState['HEALTH'] in ['OFFLINE', 'REMOVED', 'UNAVAIL']:
+        raiseState('WARNING', ret)
+    elif zpoolState['HEALTH'] in ['DEGRADED', 'FAULTED']:
+        raiseState('CRITICAL', ret)
+
+    # Check capacity and fragmentation
+    if capPercent >= args['capacity_crit'] or fragPercent >= args['fragmentation_crit']:
+        raiseState('CRITICAL', ret)
+    elif capPercent >= args['capacity_warn'] or fragPercent >= args['fragmentation_warn']:
+        raiseState('WARNING', ret)
+
+    ret['output'] = 'Pool {} is {}. Capacity is at {}%. Fragmentation is at {}%.'.format(zpoolState['NAME'],
```

Comment:
You can also put capactiy and fagmentation into details instead of output. Output should be short